0003-I-dapr-compose.md

```diff
@@ -0,0 +1,294 @@
+
+> Note: This change can be made in `dapr run` also. We can either make the change right now for 1.10 or start supporting this change from next release.
+
+> Note: This change does not impact the `bin` folder where `dapr` CLI looks for the `daprd` and `dashboard` binaries. That will still remain the same `~/.dapr/bin/` (%USERPROFILE% for windows).
+
+## Proposed Structure for `compose` configuration file
+
+> Expected default file name is `dapr.yaml`
+
+```yaml
+version: 1
+common:
+  resources_dir: ./app/components # any dapr resources to be shared across apps
+  env:  # any environment variable shared among apps
+    - DEBUG: true
+apps:
+  - app_id: webapp
+    app_dir: ./webapp/
+    resources_dir: ./webapp/components # (optional) can be default by convention too, ignore if dir is not found.
+    config_file: ./webapp/config.yaml # (optional) can be default by convention too, ignore if file is not found.
+    app_protocol: HTTP
+    app_port: 8080
+    app_health_check_path: "/healthz" # All _ converted to - for all properties defined under daprd section
+    command: ["python3" "app.py"]
+    restart: false
+  - app_id: backend
+    app_dir: ./backend/
+    app_protocol: GRPC
+    app_port: 3000
+    unix_domain_socket: "/tmp/test-socket"
+    env:
+      - DEBUG: false
+    command: ["./backend"]
+    restart: true
+```
+> Note: Running the dependencies for each app as contianers is out of scope for this discussion initially. We might consider that in the future.
+
+- Each file conatins a `common` object which contains `env`, `resources_dir` and `config_file` that can be used in common across all the apps defined in this YAML
+- There is an `apps` section that lists the different app configs.
+- Each app config has the following
+  - `app_id` application ID (mandatory field). Passed to `daprd` as `--app-id`.
+  - `app_dir` directory of the application (mandatory field).
+  - `resources_dir` (optional) directory(ies) of all dapr resources (components, resiliency policies, subscription crds) (overrides common def). Passed to `daprd` as `--resources-dir`.
+  - `config_file` (optional) the configuration file to be used for this app (overrides common def). Passed to `daprd` as `--config-file`.
+  - `app_protocol` Application protocol, HTTP, gRPC defaults to HTTP. Passed to `daprd` as `--app-protocol`.
+  - `app_port` port the app listens to if any. Passed to `daprd` as `--app-port`.
+  - other dapr run parameters (mostly pass through flags to `daprd`) All properties must have `_` as separators which will be validated(so that no unknown flags are passed) and translated to `-` for cmd line arguments for `daprd`.
+  - `command` ["exec" "arg1" "arg2"] format for application command
+  - `env` which overrides or adds to common env var defined or the shell env var passed in when `dapr compose` is called
+  - `restart` field which lets `dapr compose` know if the app/sidecar should be restarted on crash or not
```

Comment:
If we allow for automatic restarts, why would a user choose to not have it if they can?